expfmt/openmetrics_create_test.go

```diff
@@ -36,7 +36,7 @@ func TestCreateOpenMetrics(t *testing.T) {
 		in  *dto.MetricFamily
 		out string
 	}{
-		// 0: Counter, NaN as value, timestamp given.
+		// 0: Counter, NaN as value, timestamp given, no _total suffix.
```

Comment:
A counter can't have a NaN value.

Reply:
Is that part of the OpenMetrics spec or generally true?

Reply:
OpenMetrics spec, also generally true (but not enforced).

There's also limits on the size of the exemplar labels, which should probably also be implemented somewhere client side at some point so the whole scrape doesn't get dropped.

Reply:
OK, so I have done the following now:
- Changed the test.
- Added to the doc comment of `MetricFamilyToOpenMetrics` that size of exemplar labels and the value of a counter isn't checked yet.